modules/ITK/cli/statismo-build-models-utils.h

```diff
@@ -0,0 +1,175 @@
+vector<typename MeshType::Pointer> superimposeMeshes(vector<typename MeshType::Pointer> originalMeshes, typename MeshType::Pointer referenceMesh, std::set<unsigned> landmarkIndices) {
+    vector<typename MeshType::Pointer> translatedMeshes(originalMeshes.begin(), originalMeshes.end());
+    for (typename vector<typename MeshType::Pointer>::iterator it = translatedMeshes.begin(); it != translatedMeshes.end(); it++) {
+        typedef typename LandmarkBasedTransformInitializerType::LandmarkPointContainer LandmarkContainerType;
+        LandmarkContainerType movingLandmarks;
+        LandmarkContainerType fixedLandmarks;
+        typename MeshType::Pointer movingMesh = *it;
+
+        //Only use a subset of the meshes' points for the alignment since we don't have that many degrees of freedom anyways and since calculating a SVD with too many points is expensive
+        for (std::set<unsigned>::const_iterator rng = landmarkIndices.begin(); rng != landmarkIndices.end(); rng++) {
+            movingLandmarks.push_back(movingMesh->GetPoint(*rng));
+            fixedLandmarks.push_back(referenceMesh->GetPoint(*rng));
+        }
+
+        //only rotate & translate the moving mesh to best fit with the fixed mesh; there's no scaling taking place.
+        typename LandmarkBasedTransformInitializerType::Pointer landmarkBasedTransformInitializer = LandmarkBasedTransformInitializerType::New();
+        landmarkBasedTransformInitializer->SetFixedLandmarks(fixedLandmarks);
+        landmarkBasedTransformInitializer->SetMovingLandmarks(movingLandmarks);
+        typename TransformType::Pointer transform = TransformType::New();
+        transform->SetIdentity();
+        landmarkBasedTransformInitializer->SetTransform(transform);
+        landmarkBasedTransformInitializer->InitializeTransform();
+
+        typename FilterType::Pointer filter = FilterType::New();
+        filter->SetInput(movingMesh);
+        filter->SetTransform(transform);
+        filter->Update();
+
+        *it = filter->GetOutput();
+    }
+    return translatedMeshes;
+}
+
+
+template<class MeshType>
+float calculateMeshDistance(typename MeshType::Pointer mesh1, typename MeshType::Pointer mesh2) {
+    if (mesh1->GetNumberOfPoints() != mesh2->GetNumberOfPoints() || mesh1->GetNumberOfCells() != mesh2->GetNumberOfCells()) {
+        itk::ExceptionObject e(__FILE__, __LINE__, "Both meshes must have the same number of Edges & Vertices", ITK_LOCATION);
+        throw e;
+    }
+
+    float fDifference = 0;
+    for (unsigned j = 0; j < mesh1->GetNumberOfPoints(); j++) {
+        typename MeshType::PointType point1 = mesh1->GetPoint(j);
+        typename MeshType::PointType point2 = mesh2->GetPoint(j);
+        for (unsigned k = 0; k < mesh1->GetPoint(j).Size(); k++) {
+            fDifference += (point1[k] - point2[k])*(point1[k] - point2[k]);
+        }
+    }
+    fDifference /= (mesh1->GetNumberOfPoints() + MeshType::PointDimension);
```

Comment:
That should be a multiplication instead of an addition. The idea was to make the metric independent of the input's dimensionality, but it would appear that I have not opted for the right operator here at all.